src/mdns/mdns_avahi.cpp

```diff
@@ -1155,6 +1201,8 @@ void PublisherAvahi::HostSubscription::Resolve(void)
 {
     std::string fullHostName = MakeFullHostName(mHostName);
 
+    mPublisherAvahi->mHostResolutionBeginTime[mHostName] = Clock::now();
```

Comment:
Why do we have `mHostResolutionBeginTime` logic in individual mDNS implementation? 
Can we handle all `mHostResolutionBeginTime` logic in the base Publisher?

Reply:
We don't have common abstraction for subscriptions in mDNSResponder and Avahi.